### PR TITLE
feat: hey contract

### PR DESCRIPTION
**Pull request by kyranjamie**
This PR includes:
- the base contract functionality for Heystack
- a SIP-10 compliant fungible token contract called `hey`
- tested of said contract using the Clarity library
- Github CI to run the tests

These are the two main functions

```lisp
(define-public (publish-content (content (string-utf8 140)))
  (begin
    (print { content: content, index: (increment-content-index) })
    (map-set like-state 
      { content-index: (var-get content-index) } 
      { likes: 0 }
    )
    (ok { content: content, index: (var-get content-index) })))


(define-public (like-content (id int))
  (begin
    ;; cannot like content that doesn't exist
    (asserts! (>= (var-get content-index) id) (err ERR_CANNOT_LIKE_NON_EXISTENT_CONTENT))
    (ok
      (map-set like-state
        { content-index: id } 
        { likes: (+ 1 (get likes (unwrap! (get-like-count id) (err 0)))) }
      )
    )))
```

### CODE REVIEW

**Comment by aulneau** [comment]
@reedrosenbluth have interest in giving this a once over?

File: contracts/hey-token.clar [hunk 1]
```diff
@@ -0,0 +1,39 @@
+;; Implement the `ft-trait` trait defined in the `ft-trait` contract
+;; https://github.com/hstove/stacks-fungible-token
+(impl-trait 'ST3J2GVMMM2R07ZFBJDWTYEYAR8FZH5WKDTFJ9AHA.ft-trait.sip-010-trait)
+
+(define-fungible-token hey-token u100000000)
```
Comment (aulneau) [inline contracts/hey-token.clar]: do we want an upper limit?

File: contracts/hey-token.clar [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  (begin
+    (ok (ft-get-balance hey-token owner))))
+
+;; returns the total number of tokens
+(define-read-only (get-total-supply)
+  (ok (ft-get-supply hey-token)))
+
+;; returns the token name
+(define-read-only (get-name)
+  (ok "HeyStack Token"))
```
Comment (aulneau) [inline contracts/hey-token.clar]: `Heystack`

File: contracts/hey-token.clar [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  (if (is-eq tx-sender sender)
+    (begin
+      (try! (ft-transfer? hey-token amount sender recipient))
+      (print memo)
+      (ok true)
+    )
+    (err u4)))
+
+(define-public (get-token-uri)
+  (ok (some u"https://example.com")))
```
Comment (aulneau) [inline contracts/hey-token.clar]: we should do this correctly

**Review by aulneau — APPROVED** [review]

**Review by fbwoolf — APPROVED** [review]

**Pull request merged**